CorsixTH/Lua/app.lua

```diff
@@ -324,6 +323,28 @@ function App:init()
   return true
 end
 
+--! Tries to initialize the user and built in level directories, returns true on
+--! success and false on failure.
+function App:initUserLevelDir()
+  local conf_path = self.command_line["config-file"] or "config.txt"
+  self.user_level_dir = self.config.levels or
+      conf_path:match("^(.-)[^" .. pathsep .. "]*$") .. "Levels"
+
+  if self.user_level_dir:sub(-1, -1) == pathsep then
+    self.user_level_dir = self.user_level_dir:sub(1, -2)
+  end
+  if lfs.attributes(self.user_level_dir, "mode") ~= "directory" then
+    if not lfs.mkdir(self.user_level_dir) then
+       print("Notice: Level directory does not exist and could not be created.")
+       return false
+    end
+  end
+  if self.user_level_dir:sub(-1, -1) ~= pathsep then
+    self.user_level_dir = self.user_level_dir .. pathsep
+  end
+  return true
```

Comment:
It's copy / pasted from initSavegameDir below. I didn't want to expand the pr by consolidating them this late into 0.60, but they have identical functionality.